src/ripple/protocol/KnownFormats.h

```diff
@@ -42,47 +44,46 @@ class KnownFormats
     class Item
     {
     public:
-        Item (char const* name, KeyType type)
-            : m_name (name)
-            , m_type (type)
+        Item (char const* name,
+            KeyType type,
+            std::initializer_list<SOElement> uniqueFields,
+            std::initializer_list<SOElement> commonFields)
+            : soTemplate_ (uniqueFields, commonFields)
+            , name_ (name)
+            , type_ (type)
         {
-        }
-
-        Item& operator<< (SOElement const& el)
-        {
-            elements.push_back (el);
-
-            return *this;
+            // Verify that KeyType is appropriate.
+            static_assert (
+                std::is_enum<KeyType>::value ||
+                std::is_integral<KeyType>::value,
+                "KnownFormats KeyType must be integral or enum.");
         }
 
         /** Retrieve the name of the format.
         */
         std::string const& getName () const noexcept
         {
-            return m_name;
+            return name_;
         }
 
         /** Retrieve the transaction type this format represents.
         */
         KeyType getType () const noexcept
         {
-            return m_type;
+            return type_;
         }
 
-    public:
-        // VFALCO TODO make an accessor for this
-        SOTemplate elements;
+        SOTemplate const& getSOTemplate() const noexcept
```

Comment:
I don't think there's any benefit to `noexept` on functions like this, and I don't think there's ever likely to be one.  I'm fine if you want to keep it, I don't see the harm either. 

I do understand the benefit of annotating some constructors or assignments with `noexcept` for the benefit of generic code. But I don't see the benefit here. In general, if there's no benefit I'd rather leave things like this out.